src/plugins/auto/executable_network.cpp

```diff
@@ -503,7 +505,9 @@ InferenceEngine::IInferRequestInternal::Ptr MultiDeviceExecutableNetwork::Create
 
 InferenceEngine::IInferRequestInternal::Ptr MultiDeviceExecutableNetwork::CreateInferRequestImpl(InferenceEngine::InputsDataMap networkInputs,
                                                                                                 InferenceEngine::OutputsDataMap networkOutputs) {
+    _numRequestMutex.lock();
     auto num = _numRequestsCreated++;
+    _numRequestMutex.unlock();
```

Comment:
why we need the mutex? the _numRequestsCreated is already atomic

Reply:
when I reviewed this PR, I thought it's not safe for num here, I am not sure, it is like below code for me 
`lock();`
`mov eax _numRequestsCreated`
`unlock()`
`num = eax`
`lock()`
`_numRequestsCreated++`
`unlock()`

Reply:
I think this lock is completely redudnant (and would need to use lock_quard anyway) - and is not related to the subject of the PR

Reply:
Agree to separate this concern and follow up in next meeting. 

Let's merge this PR without this lock first. @myshevts

Reply:
songbell Pending
operator ++ is atomic, but assign the ++ value to num may suffer the threading issue.
I have a simple test on this, and it will sporadically run into threading issues with the value of "num", easy to reproduce when thread num increases. 
remove it from this PR, and track if further if needed